src/main/java/org/truffleruby/RubyContext.java

```diff
@@ -102,8 +102,9 @@ public RubyContext(RubyLanguage language, TruffleLanguage.Env env) {
         this.env = env;
 
         final OptionsBuilder optionsBuilder = new OptionsBuilder();
-        optionsBuilder.set(System.getProperties());
+        optionsBuilder.set(env.getOptions());
         optionsBuilder.set(env.getConfig());
+        optionsBuilder.set(System.getProperties());
```

Comment:
But does this code still prioritize command-line arguments over everything else? Can we have a test for that?